src/Services/MediaService.php

```diff
@@ -122,7 +122,7 @@ public function createFilePreview(string $filename): Model
 
         $data['name'] = $previewFilename;
         $data['link'] = Storage::url($previewFilename);
-        $data['owner_id'] = Auth::id();
+        $data['owner_id'] = Auth::check() ? Auth::id() : null;

```

Comment:
let's make some refactoring and generate preview based on the media object, it allows to use the same owner here